src/Microsoft.ML.AutoML/Utils/UserInputValidationUtil.cs

```diff
@@ -57,6 +57,14 @@ public static void ValidateNumberOfCVFoldsArg(uint numberOfCVFolds)
             }
         }
 
+        public static void ValidateSamplingKey(string samplingKeyColumnName, string groupIdColumnName, TaskKind task)
+        {
+            if (task == TaskKind.Ranking && samplingKeyColumnName != null && samplingKeyColumnName != groupIdColumnName)
+            {
+                throw new ArgumentException($"{nameof(samplingKeyColumnName)} must be the same as {nameof(groupIdColumnName)}", samplingKeyColumnName);
```

Comment:
**Discussion:** Should we throw an exception on this case, or should we simply set the `samplingKeyColumnName` of this experiment to be `groupIdColumnName` automatically? The reason why I suggest this is that because AutoML is supposed to be as automated as possible, I guess it makes sense to do this automatically.